data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorTest.java

```diff
@@ -0,0 +1,151 @@
+import org.opensearch.dataprepper.plugins.processor.configuration.KeysConfig;
+import org.opensearch.dataprepper.plugins.processor.configuration.MaxMindServiceConfig;
+import org.opensearch.dataprepper.plugins.processor.configuration.ServiceTypeOptions;
+import org.opensearch.dataprepper.plugins.processor.loadtype.LoadTypeOptions;
+import org.opensearch.dataprepper.test.helper.ReflectivelySetField;
+
+import java.net.MalformedURLException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class GeoIPProcessorTest {
+
+    public static final int REFRESH_SCHEDULE = 10;
+    public static final String SOURCE = "/peer/ip";
+    public static final String TARGET = "location";
+    public static final String PROCESSOR_PLUGIN_NAME = "geoip";
+    public static final String PROCESSOR_PIPELINE_NAME = "geoIP-processor-pipeline";
+    @Mock
+    private GeoIPProcessorService geoIPProcessorService;
+    @Mock
+    private GeoIPProcessorConfig geoCodingProcessorConfig;
+    @Mock
+    private PluginSetting pluginSetting;
+    @Mock
+    private ServiceTypeOptions serviceTypeOptions;
+    @Mock
+    private MaxMindServiceConfig maxMindServiceConfig;
+
+    @BeforeEach
+    void setUp() throws NoSuchFieldException, IllegalAccessException {
+
+        when(pluginSetting.getName()).thenReturn(PROCESSOR_PLUGIN_NAME);
+        when(pluginSetting.getPipelineName()).thenReturn(PROCESSOR_PIPELINE_NAME);
+        when(geoCodingProcessorConfig.getServiceType()).thenReturn(serviceTypeOptions);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService()).thenReturn(maxMindServiceConfig);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getLoadType())
+                .thenReturn(LoadTypeOptions.INMEMORY);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getCacheRefreshSchedule())
+                .thenReturn(Duration.ofSeconds(REFRESH_SCHEDULE));
+
+        DatabasePathURLConfig databasePathURLConfig1 = new DatabasePathURLConfig();
+        ReflectivelySetField.setField(DatabasePathURLConfig.class,
+                databasePathURLConfig1, "url", "./src/test/resources/mmdb-file/geo-lite2");
+        List<DatabasePathURLConfig> urlList = List.of(databasePathURLConfig1);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getDatabasePath()).thenReturn(urlList);
+    }
+
+    @Test
+    void doExecuteTest() throws MalformedURLException, NoSuchFieldException, IllegalAccessException {
```

Comment:
Also need a test case for tags added on match failure.

Reply:
Added test case 'test_tags_when_enrich_fails' method inside GeoIPProcessorTest file